src/zql/query/IEntityQuery.ts

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {EntitySchema} from '../schema/EntitySchema.js';
+import {IStatement} from './Statement.js';
+import {AST, Operator} from '../ast/ZqlAst.js';
+
+export type SelectedFields<T, Fields extends Selectable<any>[]> = Pick<
+  T,
+  Fields[number] extends keyof T ? Fields[number] : never
+>;
+
+export type Selectable<T extends EntitySchema> = keyof T['fields'];
+
+export type MakeHumanReadable<T> = {} & {
+  readonly [P in keyof T]: T[P] extends string ? T[P] : MakeHumanReadable<T[P]>;
+};
+
+export interface IEntityQuery<TSchema extends EntitySchema, TReturn = []> {
+  readonly select: <Fields extends Selectable<TSchema>[]>(
+    ...x: Fields
+  ) => IEntityQuery<TSchema, SelectedFields<TSchema['fields'], Fields>[]>;
+  readonly count: () => IEntityQuery<TSchema, number>;
+  readonly where: <K extends keyof TSchema['fields']>(
+    f: K,
+    op: Operator,
```

Comment:
Which operators are valid is dependent on the type of the field, and then the type for value is dependent on the type of the operator.

I don't have the full typing worked out, but I think instead of just `Operator`, we need
`NumberOperator`, `StringOperator`, `BooleanOperator`, and eventually `SubqueryOperator`.

Reply:
> Which operators are valid is dependent on the type of the field, and then the type for value is dependent on the type of the operator.

Yeah. Let me see what I can work out.